orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -353,15 +356,34 @@ public void initServiceRun(List<GatewayConfig> allGateway, Set<Node> allNodes, S
             Callable<Boolean> saltPillarRunner = saltRunner.runner(hostSave, exitCriteria, exitModel);
             saltPillarRunner.call();
 
+            // Observation: The only pillar entries which seem to change when adding new nodes are disks and nodes.
```

Comment:
True, but it might involve a lot of complexity to figuring out what pillars to save. Maybe someone introduces something dynamic in one of the pillar files and expects it to be refreshed during repair and it won't work. We're also working on updating the Salt states on existing clusters and for that we need to push down all the pillars as well again.

Reply:
Updating the salt states on existing clusters - is there a jira where I could read about this?